examples/java/src/main/java/org/apache/beam/examples/complete/game/GameStats.java

```diff
@@ -242,13 +242,20 @@ public static void main(String[] args) throws Exception {
     ExampleUtils exampleUtils = new ExampleUtils(options);
     Pipeline pipeline = Pipeline.create(options);
 
+    // Run the pipeline and wait for the pipeline to finish; capture cancellation requests from the
+    // command line.
+    PipelineResult result = runGameStats(options, pipeline);
+    exampleUtils.waitToFinish(result);
+  }
+
+  static PipelineResult runGameStats(Options options, Pipeline pipeline) {
     // Read Events from Pub/Sub using custom timestamps
     PCollection<GameActionInfo> rawEvents =
         pipeline
             .apply(
                 PubsubIO.readStrings()
-                    .withTimestampAttribute(GameConstants.TIMESTAMP_ATTRIBUTE)
-                    .fromTopic(options.getTopic()))
+                    .fromSubscription(options.getSubscription())
```

Comment:
Is there a reason we need to read from the subscription vs topic?

Reply:
Oh, If I'm not mistaken, If no subscription is provided when reading, the API will crate its own one. I'm feeding the subscription I created during the IT setup so I can delete it later.

Reply:
If it is ok, I would like to keep using the subscription created within the IT instead of the one created automatically when trying to read from the topic. Not sure why, but it looks like reading from the topic is a blocking operation and the pipeline never reaches the following step.

Reply:
But as an easy-to-use example, it should work with reading from the topic also. Can you please make it so that it can do one or the other. Reading from a subscription is more recommended for production, and good for ITs. But it is fine for development to read from a topic and will let the example be easier to try and play with.